google/api_core/client_logging.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import json
+import re
+import os
+
+LOGGING_INITIALIZED = False
+
+# TODO(<add-link>): Update Request / Response messages.
+REQUEST_MESSAGE = "Sending request ..."
+RESPONSE_MESSAGE = "Receiving response ..."
+
+# TODO(<add-link>): Update this list to support additional logging fields
+_recognized_logging_fields = ["httpRequest", "rpcName", "serviceName"] # Additional fields to be Logged.
+
+def logger_configured(logger):
+  return logger.hasHandlers() or logger.level != logging.NOTSET
+
+def initialize_logging():
+   global LOGGING_INITIALIZED
+   if LOGGING_INITIALIZED:
+     return
+   scopes = os.getenv("GOOGLE_SDK_PYTHON_LOGGING_SCOPE")
+   setup_logging(scopes)
+   LOGGING_INITIALIZED = True
+
+def parse_logging_scopes(scopes):
+  if not scopes:
+     return []
+  # TODO(<add-link>): check if the namespace is a valid namespace.
```

Comment:
My understanding was that we would check for valid scopes in the current work, and the follow-up item was to consider issuing a warning in case of an invalid scope.